go-client/session/meta_session_test.go

```diff
@@ -118,11 +118,28 @@ func TestMetaManager_FirstMetaDead(t *testing.T) {
 	for i := 0; i < 3; i++ {
 		call := newMetaCall(mm.currentLeader, mm.metas, func(rpcCtx context.Context, ms *metaSession) (metaResponse, error) {
 			return ms.queryConfig(rpcCtx, "temp")
-		})
+		}, []string{"0.0.0.0:12345", "0.0.0.0:34603", "0.0.0.0:34602", "0.0.0.0:34601"})
 		// This a trick for testing. If metaCall issue to other meta, not only to the leader, this nil channel will cause panic.
 		call.backupCh = nil
 		metaResp, err := call.Run(context.Background())
 		assert.Nil(t, err)
 		assert.Equal(t, metaResp.GetErr().Errno, base.ERR_OK.String())
 	}
 }
+
+// This case mocks the case that the server primary meta is not in the client metalist.
```

Comment:
The meta servers in the test are 0.0.0.0:3460{1..3}, which one is "not in the client metalist" ?

Reply:
When onebox starts, the primary meta server is randomized. Therefore, a loop is used, and only one meta server is passed to the go client each time. This ensures that redirection is required twice in the loop.